AppDevAnnouncements/Classes/NotificationView.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  NotificationView.swift
+//  AppDevAnnouncements
+//
+//  Created by Gonzalo Gonzalez on 10/16/19.
+//  Copyright © 2019 Cornell AppDev. All rights reserved.
+//
+
+import UIKit
+import SnapKit
+
+class NotificationView: UIView {
+
+    /// Components
+    private let topPortionView = UIView()
+    private let bottomPortionView = UIView()
+    private let dismissButton = UIButton()
+    private let visualImageView = UIImageView()
+    private let subjectLabel = UILabel()
+    private let bodyLabel = UILabel()
+    private let ctaButton = UIButton()
+
+    /// Constants
+    enum Constants {
+        ///Colors
+        static let lightGray = UIColor(displayP3Red: 245/255, green: 245/255, blue: 245/255, alpha: 1)
+        static let darkGray = UIColor(displayP3Red: 114/255, green: 114/255, blue: 114/255, alpha: 1)
+        ///Constraints
+        static let baseTopPortionViewHeight: CGFloat = 74
+        static let extraImageHeight: CGFloat = 97
+        static let dismissButtonPadding: CGFloat = 12
+        static let dismissButtonLength: CGFloat = 12
+        static let subjectLabelTopPadding: CGFloat = 32
+        static let subjectLabelHorizontalPadding: CGFloat = 24
+        static let subjectLabelHeight: CGFloat = 31
+        static let imageViewLength: CGFloat = 80
+        static let imageViewVerticalPadding: CGFloat = 16
+        static let bodyLabelTopPadding: CGFloat = 24
+        static let bodyLabelHorizontalPadding: CGFloat = 16
+        static let ctaButtonTopPadding: CGFloat = 16
+        static let ctaButtonHorizontalPadding: CGFloat = 24
+        static let ctaButtonHeight: CGFloat = 38
+        static let ctaButtonBottomPadding: CGFloat = 16
```

Comment:
Can u move this into `setupConstraints`. Its kinda bloating this part of the file

Reply:
It would be nice to move the constants into the setupConstraints where they should belong, but I'm unable to do that because NotificationViewController relies on some constants in NotificationView (to calculate totalHeight) and vice versa (to get the notification's width). The best I can do is to satisfy that and your request is to move it right on top of the setupConstraints function.